File: BeeCrowd/Simple Sort/Insertion.c
#include <stdio.h>

void insertion(int array[],int size){

    for(int i = 1; i < size; i++){
        
        int temp = array[i];
        int j = i - 1;

        while(j >= 0 && array[j] > temp){
            array[j + 1] = array[j];
            j--;
        }
        array[j + 1] = temp;
    }


}

int main(){

        int num1 [3];
        int num2 [3];

        for (int i = 0; i < 3; i++) {
            scanf("%i", &num1[i]);
            num2[i] = num1[i];
        }

        int size = sizeof(num1)/sizeof(num2[0]);

        insertion(num1,size);

        printf("%i\n",num1[0]);
        printf("%i\n",num1[1]);
        printf("%i\n",num1[2]);

        printf("\n");

        printf("%i\n",num2[0]);
        printf("%i\n",num2[1]);
        printf("%i\n",num2[2]);

}

File: BeeCrowd/Simple Sort/Insertion.java
import java.util.Scanner;
import java.util.*;

public class Insertion {

    public static void insertion(int[] array){

        for(int i = 1; i < array.length; i++){
            
            int temp = array[i];
            int j = i - 1;

            while(j >= 0 && array[j] > temp ){
                array[j + 1] = array[j];
                j--;
            }
            array[j + 1] = temp;
        }

        
    }
    public static void main(String[] args) {
        Scanner ler = new Scanner(System.in);

        int[] num1 = new int[3];
        int[] num2 = new int[3];

        for (int i = 0; i < 3; i++) {
            num1[i] = ler.nextInt();
            num2[i] = num1[i];
        }

        insertion(num1);

        System.out.println(num1[0]);
        System.out.println(num1[1]);
        System.out.println(num1[2]);

        System.out.println();

        System.out.println(num2[0]);
        System.out.println(num2[1]);
        System.out.println(num2[2]);

        ler.close();
    }
}
